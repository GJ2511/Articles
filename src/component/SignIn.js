import React from 'react';

function SignIn() {
    return (
        <div className="row">
            <div className="col-sm-9 col-md-7 col-lg-5 mx-auto">
                <div className="card card-signin my-5 bg-light ">
                    <div className="card-body">
                        <h5 className="card-title text-center">Sign In</h5>
                        <form>
                            <div className="form-group">
                                <input
                                    type="email"
                                    className="form-control rounded-pill"
                                    id="email"
                                    aria-describedby="emailHelp"
                                    placeholder="Email address"
                                    autoFocus
                                />
                            </div>
                            <div className="form-group">
                                <input
                                    type="password"
                                    className="form-control rounded-pill"
                                    id="password"
                                    placeholder="Password "
                                />
                            </div>
                            <button type="submit" className="btn btn-primary btn-lg btn-block rounded-pill">
                                Login
                            </button>
                        </form>
                    </div>
                </div>
            </div>
        </div>
    );
}

export default SignIn;
