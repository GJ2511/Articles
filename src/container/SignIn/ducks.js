import { takeLatest, put, call } from 'redux-saga/effects';

import AuthService from '../../services/authService';
import { setSignInSuccess } from '../../ducks';

const PREFIX = 'SIGN_IN';
const SIGN_IN_REQUESTED = `${PREFIX}//SIGN_IN_REQUESTED`;
const SIGN_IN_SUCCESS = `${PREFIX}//SIGN_IN_SUCCESS`;
const SIGN_IN_FAILED = `${PREFIX}//SIGN_IN_FAILED`;

const initialState = {
    isSubmitting: false,
    error: {},
};

const signInReducer = (state = initialState, action = {}) => {
    const { type, payload } = action;

    switch (type) {
        case SIGN_IN_REQUESTED:
            return {
                ...state,
                isSubmitting: true,
                error: {},
            };
        case SIGN_IN_FAILED:
            return {
                ...state,
                isSubmitting: false,
                error: payload,
            };
        default:
            return state;
    }
};

export const signInRequested = (values) => ({
    type: SIGN_IN_REQUESTED,
    payload: values,
});

function* signIn({ payload: { email, password } }) {
    try {
        const response = yield call([AuthService, 'signin'], { email, password });

        if (response.errors) {
            yield put({ type: SIGN_IN_FAILED, payload: response.errors });
        } else {
            yield put({ type: SIGN_IN_SUCCESS });
            yield put(setSignInSuccess(response.user));
        }
    } catch (error) {
        yield put({
            type: SIGN_IN_FAILED,
            payload: {
                Error: ['Something went wrong'],
            },
        });
    }
}

export function* signInSaga() {
    yield takeLatest(SIGN_IN_REQUESTED, signIn);
}

export const actions = {
    signInRequested,
};

export default signInReducer;
