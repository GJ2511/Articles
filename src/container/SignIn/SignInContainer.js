import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Formik } from 'formik';

import SignIn from '../../component/SignIn';
import { signInRequested } from './ducks';

const initialValues = {
    email: '',
    password: '',
};

class SignInContainer extends Component {
    handleValidation = (values) => {
        const errors = {};

        if (!values.email) {
            errors.email = 'Required';
        } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(values.email)) {
            errors.email = 'Invalid email address';
        }

        if (!values.password) {
            errors.password = 'Required';
        } else if (values.password.length > 20 || values.password.trim().length < 8) {
            errors.password = 'Password must contain characters between 8 - 20';
        }

        return errors;
    };

    handleSubmit = (values) => {
        this.props.signInRequested(values);
    };

    render() {
        const { error, isSubmitting } = this.props;

        return (
            <Formik
                initialValues={initialValues}
                // eslint-disable-next-line react/jsx-handler-names
                validate={this.handleValidation}
                validateOnBlur={false}
                validateOnChange={false}
                onSubmit={this.handleSubmit}
            >
                {(props) => <SignIn {...props} isSubmitting={isSubmitting} hasError={error} />}
            </Formik>
        );
    }
}

SignInContainer.propTypes = {
    error: PropTypes.object.isRequired,
    isSubmitting: PropTypes.bool.isRequired,
    signInRequested: PropTypes.func.isRequired,
};

const mapStateToProps = ({ signInReducer }) => ({
    error: signInReducer.error,
    isSubmitting: signInReducer.isSubmitting,
});

export default connect(mapStateToProps, { signInRequested })(SignInContainer);
