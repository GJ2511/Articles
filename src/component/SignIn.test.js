import React from 'react';

import SignIn from './SignIn';

const props = {
    errors: {},
    hasError: {},
    isSubmitting: false,
    touched: {},
};

describe('<SignIn tests />', () => {
    test('match snapshot', () => {
        expect(shallow(<SignIn {...props} />)).toMatchSnapshot();
    });
});
