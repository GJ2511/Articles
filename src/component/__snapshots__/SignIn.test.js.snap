// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn tests /> match snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-9 col-md-7 col-lg-5 mx-auto"
  >
    <div
      className="card card-signin my-5 bg-light "
    >
      <div
        className="card-header"
      >
        <h5
          className="card-title text-center"
        >
          Sign In
        </h5>
      </div>
      <div
        className="card-body"
      >
        <Form
          noValidate={true}
        >
          <div
            className="form-group"
          >
            <Field
              aria-describedby="emailHelp"
              autoFocus={true}
              className="form-control rounded-pill"
              id="email"
              name="email"
              placeholder="Email address"
              type="email"
            />
          </div>
          <div
            className="form-group"
          >
            <Field
              className="form-control rounded-pill"
              id="password"
              name="password"
              placeholder="Password "
              type="password"
            />
          </div>
          <button
            className="btn btn-primary btn-lg btn-block rounded-pill"
            disabled={false}
            type="submit"
          >
            Login
          </button>
        </Form>
      </div>
      <div
        className="card-footer"
      >
        Do not have an account? 
        <Link
          to="/signup"
        >
           Sign Up
        </Link>
      </div>
    </div>
  </div>
</div>
`;
